src/main/java/com/redhat/cloud/notifications/routers/BehaviorGroupMigrationService.java

```diff
@@ -49,7 +50,7 @@
      * This is used to dynamically create behavior groups names at persistence time.
      * See calling point for more details.
      */
-    private final Map<String, AtomicInteger> accountBehaviorGroupIndexes = new ConcurrentHashMap<>();
+    private final Map<BehaviorGroupIndexKey, AtomicInteger> accountBehaviorGroupIndexes = new ConcurrentHashMap<>();
```

Comment:
During the migration, behavior groups are automatically named using the following pattern: `"Behavior group " + index`. The index comes from this map and we should use a different `AtomicInteger` instance from each account <ins>and each bundle</ins>. Right now, it's only isolated for accounts and all bundles share the same index, which leads to inconsistent names for accounts with several bundles. This PR fixes that.